allauth/socialaccount/providers/openid/views.py

```diff
@@ -84,7 +85,11 @@ def login(request):
 
 @csrf_exempt
 def callback(request):
-    client = _openid_consumer(request)
+    provider = OpenIDProvider(request)
+    endpoint = request.GET.get('openid.op_endpoint', '')
+    server_settings = provider.get_server_settings(endpoint)
+    stateless = server_settings.get('stateless', False)
+    client = _openid_consumer(request, stateless)
```

Comment:
The above few lines are mostly identical the the ones further above. Can you make this more DRY by placing this inside a helper method?

Reply:
The new version based on the refactored master has less duplication. the Different source of the endpoint makes sharing much more harder.